src/main/webapp/scripts/poll.js

```diff
@@ -0,0 +1,65 @@
+ /** 
```

Comment:
Ok, so i've read through the class, take a look at the comments first, then read this. Good initial approach. Might I suggest a few adjustments? Right now you have:
class Poll {
constructor(fetchFunc, pollTime, maxAttempts);
poll();
getResult();
} 

This is ok but poll() doesnt need to be public, maxAttempts is not needed and perhaps we want to make this looks more like a cache, which is what it seems you're implementing.

class RefreshingCache {

class _CacheItem {
constructor(...);
this._key = ...;
this._updateFunc = ...;
this._value = ...;
this._setTimeoutId = ...;  // id returned by setTimeout, needed for clearTimeout

setValue(value) {
this._value = value;
}

getValue() { return this._value; }
}

constructor();

// A mapping from string to _CacheItem object
this._cache = {};

// key is an unique identifier for a cache value
// updateFunc is the function Poll will call every cadence_sec to update 
// This function puts into this._cache <key> and a new _CacheItem object
insert(key, updateFunc, cadence_sec);

// Calls getValue() on the _CacheItem object for the given key in this._cache.
get(key);

// Calls clearTimeout on the udpateFunc of the _CacheItem and removes it from this._cache. 
_remove(key);

// Calls setTimeout on the updateFunc of every _CacheItem object. After a call to this function,
// insert() cannot be called. 
start();

// Calls _remove() on every key in this._cache.
stop(); 
} 

// Actual usage.

cache = new RefreshingCache();
// Note that fetchAPIRequest is some function defined somewhere else
cache.insert("session", fetchAPIRequest, /*cadence_sec=*/10);
cache.start();
// ...
//... 
Session session = toSession(cache.get("session"));

Reply:
Maybe Danny has ideas as well

Reply:
Sorry, I probably should've waited until #11 was approved too, I think this comment would be helpful in that context better...